compiler/compiler.go

```diff
@@ -168,11 +168,8 @@ func WriteProgramCode(pkgs []*Archive, w *SourceMapFilter) error {
 		}
 	}
 
-	if _, err := w.Write([]byte("$synthesizeMethods();\nvar $mainPkg = $packages[\"" + string(mainPkg.ImportPath) + "\"];\n$packages[\"runtime\"].$init();\n$go($mainPkg.$init, []);\n$flushConsole();\n\n}).call(this);\n")); err != nil {
-		return err
-	}
-
-	return nil
+	_, err := w.Write([]byte("$synthesizeMethods();\nvar $mainPkg = $packages[\"" + string(mainPkg.ImportPath) + "\"];\n$packages[\"runtime\"].$init();\n$go($mainPkg.$init, []);\n$flushConsole();\n\n}).call(this);\n"))
```

Comment:
I don't like this simplification, I often even change my code to not do this. I prefer having the primary flow with `return nil` (all okay) at the toplevel and have `return err` only in branches for error handling. This also allows cleaner diffs when adding new stuff between this statement and the end of the function. Same goes for other such changes in this commit.